applications/coreutils/head.cpp

```diff
@@ -0,0 +1,212 @@
+#include <libsystem/cmdline/CMDLine.h>
+#include <libsystem/io/Stream.h>
+
+#define BUFFER_SIZE 1024
+
+static bool count_lines = true;
+static int count = 10;
+static bool quiet = false;
+static bool verbose = false;
+static bool zero_delimiter = false;
+static char delimiter = '\n';
+
+void set_delimiter_zero(CommandLine *cmdline, CommandLineOption *opt)
```

Comment:
this one @caelwithcats

Reply:
needs to be declared for options, or do i give the signature on top and declare it after `main`

Reply:
what do you mean?

Reply:
yes